# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/DocAccessibleParent.cpp
# Commit: 3dd1b7e892e0
# Full Hash: 3dd1b7e892e013f0909e298fff25d5c4602a27e4
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-05-11 09:42:52
# Description:
#   Bug 1768391: Invalidate the remote HyperText offsets cache when we mutate the tree. r=eeejay
#   
#   Previously, we invalidated this cache when the reorder event arrived.
#   Because the mutation and the reorder event happen in separate IPDL calls, it's possible for a client call to arrive between them.
#   If that client call queried HyperText offsets, this could result in returning incorrect information to the client or even a parent process crash.
# ==============================================================================

diff -r 7fd565174880 -r 3dd1b7e892e0 accessible/ipc/DocAccessibleParent.cpp
--- a/accessible/ipc/DocAccessibleParent.cpp	Tue May 10 22:53:56 2022 +0000
+++ b/accessible/ipc/DocAccessibleParent.cpp	Tue May 10 22:56:57 2022 +0000
@@ -296,10 +296,6 @@
   if (StaticPrefs::accessibility_cache_enabled_AtStartup()) {
     if (aEventType == nsIAccessibleEvent::EVENT_REORDER ||
         aEventType == nsIAccessibleEvent::EVENT_INNER_REORDER) {
-      if (proxy->IsHyperText()) {
-        // Invalidate the HyperText offset cache.
-        proxy->InvalidateCachedHyperTextOffsets();
-      }
       for (RemoteAccessible* child = proxy->RemoteFirstChild(); child;
            child = child->RemoteNextSibling()) {
         child->InvalidateGroupInfo();
@@ -578,13 +574,6 @@
     }
 
     remote->ApplyCache(aUpdateType, entry.Fields());
-    if (aUpdateType == CacheUpdateType::Update && remote->IsTextLeaf()) {
-      // Invalidate the HyperText offset cache.
-      RemoteAccessible* parent = remote->RemoteParent();
-      if (parent && parent->IsHyperText()) {
-        parent->InvalidateCachedHyperTextOffsets();
-      }
-    }
   }
 
   if (nsCOMPtr<nsIObserverService> obsService =